testsuite/cluster-test/src/health/fullnode_check.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::{
+    cluster::Cluster,
+    health::{HealthCheck, HealthCheckContext},
+};
+use async_trait::async_trait;
+use once_cell::sync::Lazy;
+use regex::Regex;
+use std::collections::HashMap;
+
+static VAL_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"val-(\d+)").unwrap());
+static FULLNODE_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"fn-(\d+)").unwrap());
+
+pub struct FullNodeHealthCheck {
+    cluster: Cluster,
+}
+
+impl FullNodeHealthCheck {
+    pub fn new(cluster: Cluster) -> FullNodeHealthCheck {
+        Self { cluster }
+    }
+}
+
+pub fn node_index(peer_name: String) -> String {
+    if let Some(cap) = VAL_REGEX.captures(peer_name.as_ref()) {
+        if let Some(cap) = cap.get(1) {
+            return cap.as_str().to_string();
+        }
+    }
+    if let Some(cap) = FULLNODE_REGEX.captures(peer_name.as_ref()) {
+        if let Some(cap) = cap.get(1) {
+            return cap.as_str().to_string();
+        }
+    }
+    panic!(
+        "Failed to parse peer name {} into validator_index",
+        peer_name
+    )
+}
+
+#[async_trait]
+impl HealthCheck for FullNodeHealthCheck {
+    async fn verify(&mut self, ctx: &mut HealthCheckContext) {
+        let validator = self.cluster.validator_instances();
+        let fullnode = self.cluster.fullnode_instances();
+        let mut val_latest_version = HashMap::new();
+        for v in validator {
```

Comment:
One note - it would be much more effective to query different validators in parallel.
For example, this look could be replaced with something like this:
```rust
 // This is helper function to make code below simpler
async fn get_version(instance: &Instance) -> (String, i64) {
   (instance.vaidator_index(), instance.debug_interface_client(). get_node_metric("libra_state_sync_committed_version{}").await.unwrap_or(0))
}
// ....
// Then this code can be used here to query validators in parallel:
let futures = self.cluster.validator_instances().iter().map(get_version);
// Note that join_all invokes futures in parallel
let val_latest_version = join_all(futures)?; // this wll be Vec<(String, i64)> - see return type of our get_version
let val_latest_version: HashMap<_> = val_latest_version.into_iter().collect(); // this converts vec<(key, value)> into map;
```